aten/src/ATen/native/LossMultiLabelMargin.cpp

```diff
@@ -0,0 +1,313 @@
+#include <ATen/ATen.h>
+#include <ATen/AccumulateType.h>
+#include <ATen/Dispatch.h>
+#include <ATen/TensorUtils.h>
+
+namespace at {
+namespace native {
+
+namespace {
+
+template <typename scalar_t>
+inline scalar_t multilabel_margin_loss_forward_inner_sum_cpu(
+    scalar_t* input_data,
+    int64_t* target_data,
+    scalar_t* is_target_data,
+    int64_t dim) {
+  using accscalar_t = at::acc_type<scalar_t, false>;
+  accscalar_t sum = 0;
+  for (int64_t ddt = 0; ddt < dim; ddt++) {
+    int64_t target_idx = target_data[ddt];
+    if (target_idx < 0) {
+      break;
+    }
+    is_target_data[target_idx] = 1;
+  }
+  for (int64_t dt = 0; dt < dim; dt++) {
+    int64_t target_idx = target_data[dt];
+    if (target_idx < 0) {
+      break;
+    }
+
+    scalar_t input_target = input_data[target_idx];
+    for (int64_t d = 0; d < dim; d++) {
+      if (!is_target_data[d]) {
+        scalar_t z = 1 - input_target + input_data[d];
+        if (z > 0) {
+          sum += z;
+        }
+      }
+    }
+  }
+  return sum;
+}
+
+static void multilabel_margin_loss_forward_cpu_template(
+    const Tensor& input,
+    const Tensor& target,
+    Tensor& output,
+    Tensor& is_target,
+    int64_t reduction) {
+  auto target_arg = TensorArg(target, "target", 2);
+
+  const auto ndims = input.dim();
+
+  TORCH_CHECK(
+      input.numel() > 0 && ndims <= 2,
+      "non-empty vector or matrix expected, got size: ",
+      input.sizes());
+
+  int64_t nframe, dim;
+  if (ndims <= 1) {
+    nframe = 1;
+    dim = ndims == 0 ? 1 : input.size(0);
+    TORCH_CHECK(
+        target.numel() > 0 && target.dim() <= 1 && target.numel() == dim,
+        "inconsistent size ",
+        target.sizes(),
+        " for ",
+        target_arg);
+  } else {
+    nframe = input.size(0);
+    dim = input.size(1);
+    TORCH_CHECK(
+        target.numel() > 0 && target.dim() == 2 && target.size(0) == nframe &&
+            target.size(1) == dim,
+        "inconsistent size ",
+        target.sizes(),
+        " for ",
+        target_arg);
+  }
+
+  TORCH_CHECK(
+      *target.min().data_ptr<int64_t>() >= -1, target_arg, " is out of range");
+  TORCH_CHECK(
+      *target.max().data_ptr<int64_t>() < dim, target_arg, " is out of range");
+
+  auto input_contiguous = input.contiguous();
+  auto target_contiguous = target.contiguous();
+
+  is_target.resize_as_(target);
+  TORCH_CHECK(is_target.is_contiguous(), "is_target must be contiguous");
```

Comment:
Perhaps, test this before resizing?

Reply:
I though there might be cases in which resize_as_() makes the Tensor contiguous? If that happens it would be desired.